xcode/Subconscious/Shared/Parsers/Tape.swift

```diff
@@ -91,6 +91,20 @@ struct Tape {
         }
         return false
     }
+    
+    /// Prefix-match a regular expression, and consume if match.
+    /// - Returns match object
+    mutating func consumeMatch<Pattern, Output>(
```

Comment:
Introduce `Tape.consumeMatch(_ regex:)`, which allows us to look ahead with a regex, and consume tape up until the end of the match if successful.

This single utility lets us factor away a large part of the Subtext parser.